### PR TITLE
TST: Remove yield

### PR DESCRIPTION
**Pull request by bashtage**
Remove yields in such a way that nose still works,
although at the cost of now having individual tests.  These should be
remove and the parametrized version should be enabled

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Would it be burdensome to break this into smaller PRs?  I'd be much more comfortable reviewing one file at a time.

**Comment by josef-pkt** [comment]
One problem is that the pep-8/whitespace cleanup makes the changeset much larger, than there is to actually review. But it's too late to separate that out.

File: statsmodels/datasets/tests/test_data.py [hunk 1]
```diff
@@ -1,37 +1,34 @@
...
-        assert isinstance(data, Dataset)
-        assert isinstance(df_data.data, pd.DataFrame)
-
-    def test_all_datasets(self):
-        for dataset_name in self.sets:
-            yield (self.check, dataset_name)
+# TODO: Remove when nose is completely dropped
+def test_all_datasets():
+    for set in sets:
+        test_dataset(set)
```
Comment (jbrockmendel) [inline statsmodels/datasets/tests/test_data.py]: Avoid built-in name `set`.

**Comment by bashtage** [comment]
Not that it is the most convenient, but one way to compare changesets like this is to use PyCharm's diff where you can tell it to ignore whitespace and blank line differences.

**Comment by josef-pkt** [comment]
There are at least the VAR/VECM conversion in here that is not available in @jbrockmendel's 
I try to get to the yield conversion PRs still this week

**Comment by bashtage** [comment]
Going to separate it

**Pull request closed without merging**